ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/util/ScalablePushUtilTest.java

```diff
@@ -33,25 +58,138 @@
-    when(query.getHaving()).thenReturn(Optional.empty());
-    when(query.getPartitionBy()).thenReturn(Optional.empty());
-    when(query.getRefinement())
-        .thenReturn(Optional.of(RefinementInfo.of(OutputRefinement.CHANGES)));
-    when(query.getFrom())
-        .thenReturn(new AliasedRelation(new Table(SourceName.of("Foo")), SourceName.of("blah")));
-    when(ksqlEngine.getQueriesWithSink(SourceName.of("Foo"))).thenReturn(
-        ImmutableSet.of(new QueryId("a")));
+  private Table table;
+  private Set<QueryId> queries;
+
+  @Before
+  public void setup() {
+    queries = ImmutableSet.of(queryId);
+    table = new Table(Optional.empty(), SourceName.of("asdf"));
+  }
+
+  @Test
+  public void shouldNotMakeQueryWithRowpartitionInSelectClauseScalablePush() {
+    try(MockedStatic<ColumnExtractor> columnExtractor = mockStatic(ColumnExtractor.class)) {
+      // Given:
+      givenScalablePushQuery();
+      givenSelectClause(SystemColumns.ROWPARTITION_NAME, columnExtractor);
+
+      // When:
+      final boolean isScalablePush = ScalablePushUtil.isScalablePushQuery(
+          query,
+          ksqlEngine,
+          ksqlConfig,
+          overrides
+      );
+
+      // Then:
+      assert(!isScalablePush);
+    }
+  }
+
+  @Test
+  public void shouldNotMakeQueryWithRowoffsetInSelectClauseScalablePush() {
+    try(MockedStatic<ColumnExtractor> columnExtractor = mockStatic(ColumnExtractor.class)) {
+      // Given:
+      givenScalablePushQuery();
+      givenSelectClause(SystemColumns.ROWOFFSET_NAME, columnExtractor);
+
+      // When:
+      final boolean isScalablePush = ScalablePushUtil.isScalablePushQuery(
+          query,
+          ksqlEngine,
+          ksqlConfig,
+          overrides
+      );
+
+      // Then:
+      assert(!isScalablePush);
+    }
+  }
+
+  @Test
+  public void shouldNotMakeQueryWithRowpartitionInWhereClauseScalablePush() {
+    try(MockedStatic<ColumnExtractor> columnExtractor = mockStatic(ColumnExtractor.class)) {
+      // Given:
+      givenScalablePushQuery();
+      givenWhereClause(SystemColumns.ROWPARTITION_NAME, columnExtractor);
+
+      // When:
+      final boolean isScalablePush = ScalablePushUtil.isScalablePushQuery(
+          query,
+          ksqlEngine,
+          ksqlConfig,
+          overrides
+      );
+
+      // Then:
+      assert(!isScalablePush);
+    }
+  }
+
+  @Test
+  public void shouldNotMakeQueryWithRowoffsetInWhereClauseScalablePush() {
+    try(MockedStatic<ColumnExtractor> columnExtractor = mockStatic(ColumnExtractor.class)) {
+      // Given:
+      givenScalablePushQuery();
+      givenWhereClause(SystemColumns.ROWOFFSET_NAME, columnExtractor);
+
+      // When:
+      final boolean isScalablePush = ScalablePushUtil.isScalablePushQuery(
+          query,
+          ksqlEngine,
+          ksqlConfig,
+          overrides
+      );
+
+      // Then:
+      assert(!isScalablePush);
+    }
+  }
+
+  @Test
+  public void shouldMakeCompatibleQueryScalablePush() {
```

Comment:
I don't think we need this test. Conceptually it's the same as `isScalablePushQuery_true()`. It's nice to know that the `ColumnExtractor` mock doesn't break the testing but once we've confirmed that for ourselves, I don't think we need to check this in.

Reply:
Maybe we should bring in @AlanConfluent to talk about this since it's his code.

Reply:
Alan's welcome to chime in but I think the question of whether this newly added test is valuable is separate from the test coverage he added in his PR? Or maybe I'm misunderstanding the question.

Reply:
Sorry, I didn't recognize this was my own test.. You're right.